Dockerfile

```
@@ -0,0 +1,54 @@
+# GPTQ-for-LLaMa and Text Generation WebUI Dockerfile
+FROM nvidia/cuda:11.8.0-devel-ubuntu22.04 as builder
+
+RUN apt-get update && \
+    apt-get install --no-install-recommends -y git build-essential python3-dev python3-pip && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN --mount=type=cache,target=/root/.cache/pip pip3 install torch torchvision torchaudio
```

```suggestion
RUN --mount=type=cache,target=/root/.cache/pip pip3 install --upgrade pip setuptools
RUN --mount=type=cache,target=/root/.cache/pip pip3 install torch torchvision torchaudio
```

Reinstall `pip` to avoid using the system `pip` package. The `pip3` installed by `apt install python3-pip` is placed in `/usr/bin/pip3`. "Upgrade" it with `pip3 install --upgrade pip` will install a new one in `/usr/bin/local/pip3`.

---

what is bad about the system's pip? it seems to add an extra 200 seconds on the buildtime

without upgrading pip:
 => [stage-1  8/15] RUN . /app/venv/bin/activate &&     pip3 install torch torchvision torchaudio &&     pip3   491.5s
 => [builder 7/8] RUN . /build/venv/bin/activate &&     pip3 install torch torchvision torchaudio &&     pip3   484.5s

with upgrading pip:
=> [builder 7/8] RUN . /build/venv/bin/activate &&     pip3 install --upgrade pip set  628.4s
 => [stage-1  8/15] RUN . /app/venv/bin/activate &&     pip3 install --upgrade pip set  684.6s

---

what is bad about the system pip? it seems to increase the build time by ~ 200 seconds